Core/Core/Grades/View/GradeListView.swift

```diff
@@ -377,31 +401,120 @@ public struct GradeListView: View, ScreenViewTrackable {
         userID: String
     ) -> some View {
-        ForEach(assignmentSections) { section in
-            Section(header:
-                VStack(spacing: 2) {
-                    HStack(spacing: 0) {
-                        Text(section.title ?? "")
-                            .foregroundStyle(Color.textDark)
-                            .font(.semibold14)
-                            .padding(.horizontal, 16)
-                        Spacer()
+        List {
+            ForEach(assignmentSections) { section in
+                Section(header: listSectionView(title: section.title)) {
+                    ForEach(section.assignments) { assignment in
+                        listRowView(
+                            assignment: assignment,
+                            userID: userID,
+                            courseColor: courseColor
+                        )
                     }
-                    .frame(height: 51)
-                    Divider()
                 }
-                .background(Color.backgroundLight)
+                .listSectionSeparator(.hidden)
+            }
+        }
+        .background(Color.backgroundLightest)
+        .iOS16HideListScrollContentBackground()
+        .listStyle(.plain)
+    }
+
+    @ViewBuilder
+    private func listSectionView(title: String?) -> some View {
+        VStack(spacing: 2) {
+            HStack(spacing: 0) {
+                Text(title ?? "")
+                    .foregroundStyle(Color.textDark)
+                    .font(.semibold14)
+                    .padding(.horizontal, 16)
+                Spacer()
+            }
+        }
+        .frame(height: 35)
+        .padding(.horizontal, -16)
+    }
+
+    @ViewBuilder
+    private func listRowView(
+        assignment: Assignment,
+        userID: String,
+        courseColor _: UIColor?
+    ) -> some View {
+        Button {
+            viewModel.didSelectAssignment.accept((viewController, assignment))
+        } label: {
+            GradeRowView(
+                assignment: assignment,
+                userID: userID,
+                isWhatIfScoreEnabled: viewModel.isWhatIfScoreEnabled
             ) {
-                LazyVStack(spacing: 0) {
-                    ForEach(section.assignments) { assignment in
-                        Button {
-                            viewModel.didSelectAssignment.accept((viewController, assignment))
-                        } label: {
-                            GradeRowView(assignment: assignment, userID: userID)
-                        }
-                        .buttonStyle(ContextButton(contextColor: courseColor))
-                        Divider()
+                isScoreEditorPresented.toggle()
+            }
+        }
+        .listRowInsets(EdgeInsets())
+        .iOS16RemoveListRowSeparatorLeadingInset()
+        .swipeActions(edge: .trailing) { revertWhatIfScoreSwipeButton }
+//        .buttonStyle(ContextButton(contextColor: courseColor))
```

Comment:
buttonStyle doesn't play nicely with list row. If buttonStyle is added, additional buttons can't be pressed.